codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/AsyncClientBuilderClass.java

```diff
@@ -54,12 +54,24 @@ public TypeSpec poetSpec() {
                          .addJavadoc("Internal implementation of {@link $T}.", builderInterfaceName);
 
         if (model.getEndpointOperation().isPresent()) {
-            builder.addMethod(enableEndpointDiscovery());
+            builder.addMethod(endpointDiscoveryEnabled())
+                   .addMethod(enableEndpointDiscovery());
         }
 
         return builder.addMethod(buildClientMethod()).build();
     }
 
+    private MethodSpec endpointDiscoveryEnabled() {
+        return MethodSpec.methodBuilder("endpointDiscoveryEnabled")
+                         .addAnnotation(Override.class)
+                         .addModifiers(Modifier.PUBLIC)
+                         .returns(builderClassName)
+                         .addParameter(boolean.class, "endpointDiscoveryEnabled")
+                         .addStatement("this.endpointDiscoveryEnabled = endpointDiscoveryEnabled")
+                         .addStatement("return this")
+                         .build();
+    }
+
     private MethodSpec enableEndpointDiscovery() {
         return MethodSpec.methodBuilder("enableEndpointDiscovery")
```

Comment:
Deprecate here as well? Can we also remove it for new services?

Reply:
I will deprecate here as well. Do you have a strategy for removing it for new services? How do we identify 'new services'? One strategy is just to say 'service != DDB' which seems yuck.

Reply:
We could also add a customization: "enableEndpointDiscoveryMethodRequired" and just enable it for DynamoDB.